src/nvenc/nvenc_base.cpp

```diff
@@ -576,20 +600,22 @@ namespace nvenc {
         nvenc_status_case(NV_ENC_ERR_RESOURCE_REGISTER_FAILED);
         nvenc_status_case(NV_ENC_ERR_RESOURCE_NOT_REGISTERED);
         nvenc_status_case(NV_ENC_ERR_RESOURCE_NOT_MAPPED);
-        // Newer versions of sdk may add more constants, look for them the end of NVENCSTATUS enum
+        // Newer versions of sdk may add more constants, look for them at the end of NVENCSTATUS enum
 #undef nvenc_status_case
         default:
           return std::to_string(status);
       }
     };
 
-    last_error_string.clear();
+    last_nvenc_error_string.clear();
     if (status != NV_ENC_SUCCESS) {
+      /* This API function gives broken strings more often than not
       if (nvenc && encoder) {
-        last_error_string = nvenc->nvEncGetLastErrorString(encoder);
-        if (!last_error_string.empty()) last_error_string += " ";
+        last_nvenc_error_string = nvenc->nvEncGetLastErrorString(encoder);
+        if (!last_nvenc_error_string.empty()) last_nvenc_error_string += " ";
       }
-      last_error_string += status_string(status);
+      */
+      last_nvenc_error_string += status_string(status);

```

Comment:
Instead of commenting out the code, should it just be removed?

Reply:
It's still a valid code from our end and can help figuring out potential errors.
For some functions it works, for some it doesn't.
It's just not safe to enable by default because the gibberish it sometimes produce smells of out-of-bounds reads.

Reply:
Sorry, I think I caused some confusion by including L618. I am referring to L612-L617, which is just a comment.

```cpp
      /* This API function gives broken strings more often than not
      if (nvenc && encoder) {
        last_nvenc_error_string = nvenc->nvEncGetLastErrorString(encoder);
        if (!last_nvenc_error_string.empty()) last_nvenc_error_string += " ";
      }
      */
```

Reply:
There's this API function
```cpp
// NvEncGetLastErrorString
/**
 * \brief Get the description of the last error reported by the API.
 *
 * This function returns a null-terminated string that can be used by clients to better understand the reason
 * for failure of a previous API call.
 *
 * \param [in] encoder
 *   Pointer to the NvEncodeAPI interface.
 *
 * \return
 *   Pointer to buffer containing the details of the last error encountered by the API.
 */
const char * NVENCAPI NvEncGetLastErrorString          (void* encoder);
```
Sometimes it works, sometimes it doesn't. When it works it's helpful, when it doesn't it's dangerous.
If I remove it we lose the potentially helpful part, if I keep it it's still dangerous.
So I compromised and commented it out.